### PR TITLE
get old and new projects from git

### PR DESCRIPTION
**Pull request by tschuerl**
Run typeCart on different commits of the same project. To do this, we need to download or checkout these versions from git or github. 

[Explanation](https://quip-amazon.com/0rkQA95WT3VW/Lib2Git-Package) of methods used from external NuGet package **Lib2Git**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: TypeInjections/Tool/Tool.fs [hunk 1]
```diff
@@ -4,39 +4,88 @@
...
               failwith "usage: program -r GITHUB_URL OLD_COMMIT NEW_COMMIT OUTPUTFOLDER"
             
-            Console.WriteLine("***** remote github")
+            let url = argvList.Item(1)
+            
+            let commit1 = argvList.Item(2)
+            let commit2 = argvList.Item(3)
+            
+            let path = argvList.Item(4)
+            let pathOld = path + "/Old"
```
Comment (mschlaipfer) [inline TypeInjections/Tool/Tool.fs]: this can be `$"{path}/Old"`

File: TypeInjections/Tool/Tool.fs [hunk 2]
```diff
@@ -4,39 +4,88 @@
...
+open LibGit2Sharp
 open System.IO
 open System
 open TypeInjections
 
 module Tool =
     
+    //NOTE 'location' refers to either the github url OR path to repo on local computer
+    let getCommit (SHA: string) (location:string) (outputPath:string) : unit =
+      Utils.log ("path to repo is " + outputPath)
```
Comment (mschlaipfer) [inline TypeInjections/Tool/Tool.fs]: this can be `Utils.log $"path to repo is {outputPath}"`

File: TypeInjections/Tool/Tool.fs [hunk 3]
```diff
@@ -4,39 +4,88 @@
...
 
+open LibGit2Sharp
 open System.IO
 open System
 open TypeInjections
 
 module Tool =
     
+    //NOTE 'location' refers to either the github url OR path to repo on local computer
+    let getCommit (SHA: string) (location:string) (outputPath:string) : unit =
```
Comment (mschlaipfer) [inline TypeInjections/Tool/Tool.fs]: for `getCommit` I'm a bit surprised this has return type `unit`

File: TypeInjections/Tool/Tool.fs [hunk 4]
```diff
@@ -4,39 +4,88 @@
...
+            
+            let commit1 = argvList.Item(2)
+            let commit2 = argvList.Item(3)
+            
+            let path = argvList.Item(4)
+            let pathOld = path + "/Old"
+            let pathNew = path + "/New"
+          
+            // lib2git requires an empty folder to place repos
+            for a in [pathOld;pathNew] do
```
Comment (mschlaipfer) [inline TypeInjections/Tool/Tool.fs]: did you run the formatter on this file?

File: TypeInjections/Tool/Tool.fs (hunk 3 above)
Comment (tschuerl) [reply TypeInjections/Tool/Tool.fs]: Yeah the name is misleading, I renamed method to `checkoutCommit`

File: TypeInjections/Tool/Tool.fs (hunk 4 above)
Comment (tschuerl) [reply TypeInjections/Tool/Tool.fs]: Now file is formatted

File: TypeInjections/Tool/Tool.fs (hunk 1 above)
Comment (tschuerl) [reply TypeInjections/Tool/Tool.fs]: Fixed

File: TypeInjections/Tool/Tool.fs (hunk 2 above)
Comment (tschuerl) [reply TypeInjections/Tool/Tool.fs]: Fixed

File: TypeInjections/Tool/Tool.fs [hunk 5]
```diff
@@ -1,53 +1,97 @@
...
 open System.IO
 open System
 open TypeInjections
 
 module Tool =
-    
+
+    //NOTE 'location' refers to either the github url OR path to repo on local computer
+    let checkoutCommit (SHA: string) (location: string) (outputPath: string) : unit =
+        Utils.log $"path to repo is {outputPath}"
```
Comment (ruijiefang) [inline TypeInjections/Tool/Tool.fs:15]: CI is giving warnings here, looks like a simple fix?

Comment (ruijiefang) [reply TypeInjections/Tool/Tool.fs:15]: I can give approve if this is blocking you @tschuerl

Comment (tschuerl) [reply TypeInjections/Tool/Tool.fs:15]: Yeah the warning is about the variable `SHA` being in all capital letters, I can just change the variable name

**Review by ruijiefang — APPROVED** [review]

**Pull request merged**